javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -132,11 +151,15 @@ final class Formatter(private val dest: Appendable) extends Closeable with Flush
 
           def pad(argStr: String, prefix: String = "",
                   preventZero: Boolean = false) = {
-            val prePadLen = argStr.length + prefix.length
+            val s = 
+              if (hasFlag(","))
+                formatThousands(argStr)
```

Comment:
You shouldn't `formatThousands` in `pad`, just because `pad` is used in the places where we also need formatting (which is probably not true... what about "xoA"?)